### PR TITLE
Ugliness necessary to prevent the extra block fetch from worker

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

**Review by coburncoburn — COMMENTED** [review]
this is just to prevent the literal fetching of it from infura twice, but doesn't actually change the output of this function, right?

**Comment by jflatow** [comment]
> this is just to prevent the literal fetching of it from infura twice, but doesn't actually change the output of this function, right?

Yeah it doesn't change what the blocks we process in the runtime are, but changes what blocks we fetch and when

File: pallets/cash/src/internal/events.rs [hunk 1]
```diff
@@ -102,29 +102,41 @@ pub fn track_chain_events_on<T: Config>(chain_id: ChainId) -> Result<(), Reason>
...
         let event_queue = get_event_queue::<T>(chain_id)?;
         let slack = queue_slack(&event_queue) as u64;
-        let blocks = fetch_chain_blocks(
-            chain_id,
-            last_block.number() + 1,
-            last_block.number() + 1 + slack,
-            starport,
-        )?
-        .filter_already_supported(&me.substrate_id, pending_blocks);
+        let blocks = next_block.concat(
```
Comment (coburncoburn) [inline pallets/cash/src/internal/events.rs:118]: feels like we should at least verify the parent_hash chain when fetching here, similar to the way re-orgs do. Am I missing something?

Comment (jflatow) [reply pallets/cash/src/internal/events.rs:118]: The reorgs don't verify, the chain is ultimately checked by the runtime

**Pull request closed without merging**